parsers/NTESMO.py

```diff
@@ -108,16 +126,20 @@ def extract_demand_price_data(file: bytes) -> pd.DataFrame:
 
 def get_data(
     session: Session,
-    target_datetime: datetime,
     extraction_func: Callable[[bytes], pd.DataFrame],
-    logger: Logger,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
```

Comment:
There should be no need for the default arguments here. The get data function should get these passed from the calling functions.

Reply:
Resolves in d4d8fbf